### DISCUSSION TITLE
Вопрос о записи версии кода

### DISCUSSION THREAD

**Discussion by MadCat9958** [MadCat9958/MadBotPublic#26] [Q&A]
Вчера я, переписывая снова код, задался вопросом: а куда всё-же записывать текущую версию бота и как?

На данный момент, используется запись в `.env` файле. 
### Плюсы:
- Можно для себя пометить, с какой версией работаешь на данный момент.
- Свободная запись версии бота. Можно хоть `aboba` написать.

### Минусы:
- Кто-то, работая с текущим кодом, не будет видеть текущую версию, не заходя на GitHub или Discord. Да и как понять, какая сейчас версия? Вдруг `0.0.1`?
  - Да, в `.env_example` стоит "текущая" версия как placeholder. Но его могут с лёгкостью забыть изменить, так как ничего не поменяется от этого.
- Свободная запись версии бота. Видеть `aboba` в версии бота не очень хочется (если только не на 1 апреля).
- Невозможность запросто получить текущие подверсии. Сейчас не используется какой-либо класс, который разбивает версии на мажорную, минорную и на микро-версию. Про всякие альфа, бета, гамма, штрихи вообще можно забыть. А вдруг ещё будет `aboba`? Будет отвал башки (особенно, если нет проверки, что версия придерживается стандарта `x.x.x`). 

## Мои предложения
- Попробовать сделать внутри файла конфига класс текущей версии бота.
- Попробовать сделать внутри файла конфига строку текущей версии бота.

Прошу каждого, кто просмотрит это обсуждение и у кого есть хоть что-то в качестве идеи или предложения, ответить сюда со своим взглядом на этот вопрос. Спасибо.

**Comment by kingvlad1** [comment]
Зачем их вобще нумеровать? ⚡

**Reply by MadCat9958** [reply]
Rolling-release? Мне кажется, обычным пользователям проще будет понимать, какая работа проделана, видя новую циферку версии.

**Comment by MadCat9958** [comment]
Создать 6 классов для определения версии

:copyright: попизделки за столом
